### ISSUE TITLE
Compilation fails in test

### ISSUE THREAD

**Issue by AladW** [jcnelson/fskit#9]
Hi,

I'm trying to compile fskit on Arch Linux, but get the following error on "make all"
```
g++ -march=i686 -mtune=generic -O2 -pipe -fstack-protector-strong --param=ssp-buffer-size=4 -o test-route.o  -I../include -I. -c test-route.cpp -D_REENTRANT -D_THREAD_SAFE -D__STDC_FORMAT_MACROS -D_FILE_OFFSET_BITS=64
test-route.cpp: In function ‘int main(int, char**)’:
test-route.cpp:145:87: error: invalid conversion from ‘int (*)(fskit_core*, fskit_route_metadata*, fskit_entry*, fskit_dir_entry**, uint64_t) {aka int (*)(fskit_core*, fskit_route_metadata*, fskit_entry*, fskit_dir_entry**, long long unsigned int)}’ to ‘fskit_entry_route_readdir_callback_t {aka int (*)(fskit_core*, fskit_route_metadata*, fskit_entry*, fskit_dir_entry**, unsigned int)}’ [-fpermissive]
    readdir_rh = fskit_route_readdir( &core, "/test-dir", readdir_cb, FSKIT_SEQUENTIAL );
                                                                                       ^
In file included from ../include/fskit/fskit.h:52:0,
                 from common.h:25,
                 from test-route.h:25,
                 from test-route.cpp:22:
../include/fskit/route.h:194:5: note: initializing argument 3 of ‘int fskit_route_readdir(fskit_core*, const char*, fskit_entry_route_readdir_callback_t, int)’
 int fskit_route_readdir( struct fskit_core* core, char const* route_regex, fskit_entry_route_readdir_callback_t readdir_cb, int consistency_discipline );
     ^
Makefile:25: recipe for target 'test-route.o' failed
make[1]: *** [test-route.o] Error 1
rm test-mknod.o test-write.o test-open.o
make[1]: Leaving directory '/home/archie/src/fskit/src/fskit/test'
Makefile:5: recipe for target 'all' failed
make: *** [all] Error 2
```
See http://pastie.org/private/5yr8pyemhtnvsryxapopa for the complete log

Dependency versions:

libstdc++5 3.3.6-5
glibc 2.21-3
fuse 2.9.3-2
gcc 4.9.2-4

Regards,
Alad

**Comment by jcnelson** [comment]
Hi Alad, thank you for reporting this. 

There is a type mismatch in the test for the "num_dents" argument in the fskit_entry_route_readdir_callback_t type definition:  in the test, the implementation uses uint64_t, but fskit_entry_route_readdir_callback_t requires size_t.  I had not noticed it before because on x86_64 (my host's architecture), these types have the same size, and thus the compiler does not complain.

This has been corrected as of e626604224bda23db237ba36aaabaf2baacae099.  Can you do a git pull and confirm that it's gone?

Thanks!
-Jude

**Comment by AladW** [comment]
Yep, I can confirm it builds correctly. Cheers for the quick response :)

Alad
